src/web/api/src/main/java/org/locationtech/geogig/rest/geopkg/GeoPkgExportWebOp.java

```diff
@@ -0,0 +1,215 @@
+        return "Description coming, check back soon!";
+    }
+
+    private ObjectId createFeatureTypeSchema(JDBCDataStore dataStore, String srcPath,
+        String targetTable, Form options) throws IOException {
+
+        ObjectId featureTypeId = null;
+        // see if the dataStore already has the target table
+        if (!Arrays.asList(dataStore.getTypeNames()).contains(targetTable)) {
+            // table doesn't exist, let's create it.
+            // need the geogig context
+            final Request request = getRequest();
+            final Context context = super.getContext(request);
+            // get any specified featureType filter
+            String sFeatureTypeId = options.getFirstValue(FEATURE_TYPE_PARAM, null);
+            SimpleFeatureType outputFeatureType;
+            if (sFeatureTypeId != null) {
+                // Check the feature type id string is a correct id
+                Optional<ObjectId> id = context.command(RevParse.class)
+                    .setRefSpec(sFeatureTypeId).call();
+                checkParameter(id.isPresent(), "Invalid feature type reference", sFeatureTypeId);
+                RevObject.TYPE type = context.command(ResolveObjectType.class).setObjectId(id.get())
+                    .call();
+                checkParameter(type.equals(RevObject.TYPE.FEATURETYPE),
+                    "Provided reference does not resolve to a feature type: ", sFeatureTypeId);
+                outputFeatureType = (SimpleFeatureType) context
+                    .command(RevObjectParse.class).setObjectId(id.get())
+                    .call(RevFeatureType.class).get().type();
+                featureTypeId = id.get();
+            } else {
+                // no specified featureType, get it from the source
+                try {
+                    final Repository repository = context.repository();
+                    SimpleFeatureType sft = getFeatureType(srcPath, repository);
+                    outputFeatureType = new SimpleFeatureTypeImpl(new NameImpl(targetTable),
+                        sft.getAttributeDescriptors(), sft.getGeometryDescriptor(),
+                        sft.isAbstract(), sft.getRestrictions(), sft.getSuper(),
+                        sft.getDescription());
+                } catch (GeoToolsOpException e) {
+                    throw new RuntimeException("No features to export.", e);
+                }
+            }
+            try {
+                dataStore.createSchema(outputFeatureType);
+            } catch (IOException e) {
+                throw new RuntimeException("Cannot create new table in database", e);
+            }
+        }
+        return featureTypeId;
+    }
+
+    private SimpleFeatureType getFeatureType(final String sourceTreeIsh, final Repository repository) {
+
+        checkParameter(sourceTreeIsh != null, "No path specified.");
+
+        final String headTreeish;
+        final String featureTreePath;
+        final Optional<ObjectId> rootTreeId;
```

Comment:
again, copied method. This actually exists in a couple places, so this is the 3rd copy of the same code.